### DISCUSSION TITLE
How does MFIter work when a MultiFab has ghost cells

### DISCUSSION THREAD

**Discussion by ZiyangHuang** [AMReX-Codes/amrex#3372] [Q&A] [answered]
Suppose I declare MFa a MultFab with 2 ghost cells. When I use: 
for (MFIter mfi(MFa, TilingIfNotGPU()); mfi.isValid(); ++mfi)
{
            const Box& bx = mfi.tilebox();

            amrex::ParallelFor(bx,
            [=] AMREX_GPU_DEVICE (int i, int j, int k)
            {
                    ....
            });       
}

Are the ghost cells of MFa included in the above loop?

**Comment by WeiqunZhang** [comment, marked as answer]
No, the ghost cells are not included in the loop above. If you want to include ghost cells, you can use `mfi.growntilebox()`. By default, `growntilebox` includes ghost cells. You can also do `growntilbox(1)` if you only want one ghost cell. You can also do `growntilebox(IntVect(0,1,0))` for zero ghost cells in x and z-directions and one ghost cell in y-direction.
